sublack.py

```diff
@@ -287,6 +292,50 @@ def run(self, edit):
         Black(self.view)(edit, extra=["--diff"])
 
 
+class BlackToggleBlackOnSaveCommand(sublime_plugin.TextCommand):
+    """
+    The "black_toggle_black_on_save" switches the setting with the same
+    name temporarily per view.
+    """
+
+    def is_enabled(self):
+        return is_python(self.view)
+
+    is_visible = is_enabled
+
+    def description(self):
+        settings = get_settings(self.view)
+        if settings['black_on_save']:
+            return "Sublack: Disable black on save"
+        else:
+            return "Sublack: Enable black on save"
+
+    def run(self, edit):
+        view = self.view
+
+        settings = get_settings(view)
+        current_state = settings['black_on_save']
+        next_state = not current_state
+
+        # A setting set on a particular view overules all other places where
+        # the same setting could have been set as well. E.g. project settings.
+        # Now, we first `erase` such a view setting which is luckily an
+        # operation that never throws, and immediately check again if the
+        # wanted next state is fulfilled by that side effect.
+        # If yes, we're almost done and just clean up the status area.
+        view.settings().erase(BLACK_ON_SAVE_VIEW_SETTING)
+        if get_settings(view)['black_on_save'] == next_state:
+            view.erase_status(STATUS_KEY)
+            return
+
+        # Otherwise, we set the next state, and indicate in the status bar
+        # that this view now deviates from the other views.
+        view.settings().set(BLACK_ON_SAVE_VIEW_SETTING, next_state)
+        view.set_status(
+            STATUS_KEY, "black: {}".format('ON' if next_state else 'OFF')
```

Comment:
the status do not change for already "deviating view" if we add or remove  sublack.black_on_save in project settings.
If it's an easy fix you can go for it not, we'll let it as it is

Thank for you work !!

Reply:
You're correct but I thought this is a good enough iteration for now.

A correct solution instead of this 80% approximation would be non trivial bc we had to listen for settings changes. I think it's too much effort for the value right now. 

If I run into this visual bug often, I will probably do another PR.

Reply:
it's ok for me